graphviz/files.py

```diff
@@ -143,7 +144,14 @@ def save(self, filename=None, directory=None):
         if directory is not None:
             self.directory = directory
 
-        filepath = self.filepath
+        if self.filename is None and self.directory is None and cleanup:
+            nmdtmpfile = tempfile.NamedTemporaryFile(suffix='.gv', delete=False)
+            nmdtmpfile.close()
```

Comment:
The conentional use case for `NamedTemporaryFile` would be to use it directly as a file object (i.e. to use its `.write()`-method).